src/components/keyboard.js

```diff
@@ -0,0 +1,438 @@
+import TextArea from './text-area';
+import Key from './key';
+
+export default class Keyboard {
+  constructor(selectorStr) {
+    this.text = new TextArea();
+
+    this.langList = ['en', 'ru'];
+    this.setLang();
+    this.setDefaultControlKeyState();
+
+    this.keys = [];
+    this.keyMap = {};
+    this.selector = document.querySelector(selectorStr);
+
+    this.buildKeyBoard();
+    this.addEventListeners();
+  }
+
+  setLang(langIndex) {
+    if (typeof window.localStorage.lang === 'undefined') {
+      window.localStorage.lang = 0;
+    }
+
+    if (langIndex === undefined) {
+      this.currentLang = Number(window.localStorage.lang);
+    } else {
+      this.currentLang = langIndex;
+    }
+
+    window.localStorage.lang = this.currentLang;
+  }
+
+  buildKeyBoard() {
+    const keyboard = document.createElement('div');
+    keyboard.classList.add('keyboard');
+
+    keyboard.append(this.text.element);
+
+    const keys = document.createElement('div');
+    keys.classList.add('keys');
+
+    this.createKeys(keys);
+
+    keyboard.append(keys);
+    keyboard.append(Keyboard.renderLegend());
+    this.selector.append(keyboard);
+  }
+
+  static renderLegend() {
+    const legend = document.createElement('p');
+    legend.classList.add('legend');
+
+    legend.innerHTML = '<b>Ctrl + Space</b> - change language';
+    return legend;
+  }
+
+  addEventListeners() {
+    document.addEventListener('keydown', (e) => {
+      e.preventDefault();
```

Comment:
Shouldn't you prevent default only for items that are in your map? I cannot use F1..12 key

Reply:
Previously, I planned to make all keys)
I agree, better remove, because it is inconvenient to open for example DevTools, etc.